data/crac/crac-util/src/main/java/com/farao_community/farao/data/crac_util/CracAliasesUtil.java

```diff
@@ -0,0 +1,72 @@
+
+    private CracAliasesUtil() {
+    }
+
+    public static void createAliases(Crac crac, Network network, UcteNodeMatchingRule rule) {
+        // List (without duplicates) all the crac elements that need to be found in the network
+        Set<String> elementIds = new HashSet<>();
+        crac.getCnecs().forEach(cnec -> elementIds.add(cnec.getNetworkElement().getId()));
+        crac.getContingencies().forEach(contingency -> handleAliases(contingency.getNetworkElements(), elementIds));
+        crac.getNetworkActions().forEach(networkAction -> handleAliases(networkAction.getNetworkElements(), elementIds));
+        crac.getRangeActions().forEach(rangeAction -> handleAliases(rangeAction.getNetworkElements(), elementIds));
+
+        // Try to find a corresponding element in the network, and add elementId as an alias
+        elementIds.forEach(elementId -> {
+            Optional<Identifiable<?>> correspondingElement = network.getIdentifiables().stream().filter(identifiable -> anyMatch(identifiable, elementId, rule)).findAny();
+            correspondingElement.ifPresent(identifiable -> identifiable.addAlias(elementId));
+        });
+    }
+
+    private static void handleAliases(Set<NetworkElement> networkElements, Set<String> elementIds) {
+        networkElements.forEach(networkElement -> elementIds.add(networkElement.getId()));
+    }
+
+    /* It only works correctly for Cnec with direction = both. This corrupts the other Cnec.
+    TODO : only look for the elements with reverse ID if all the thresholds of the cnec have a BOTH direction.
+     */
+    private static boolean anyMatch(Identifiable<?> identifiable, String cnecId, UcteNodeMatchingRule rule) {
+        return nameMatches(identifiable, cnecId, rule, false) ||
+            aliasMatches(identifiable, cnecId, rule, false) ||
+            nameMatches(identifiable, cnecId, rule, true) ||
+            aliasMatches(identifiable, cnecId, rule, true);
+    }
+
+    private static boolean nameMatches(Identifiable<?> identifiable, String cnecId, UcteNodeMatchingRule rule, boolean reverse) {
+        return identifiable.getId().trim().matches(checkWithPattern(cnecId, rule, reverse));
+    }
+
+    private static boolean aliasMatches(Identifiable<?> identifiable, String cnecId, UcteNodeMatchingRule rule, boolean reverse) {
+        return identifiable.getAliases().stream().anyMatch(alias -> alias.trim().matches(checkWithPattern(cnecId, rule, reverse)));
+    }
+
+    private static String checkWithPattern(String string, UcteNodeMatchingRule rule, boolean reverse) {
+        int first = reverse ? POSITION_OF_SECOND_UCTE_NODE : 0;
+        int second = reverse ? 0 : POSITION_OF_SECOND_UCTE_NODE;
+        return Pattern.quote(string.substring(first, first + rule.getNbCharacter())) + ".*" + " " + Pattern.quote(string.substring(second, second + rule.getNbCharacter())) + ".*" + Pattern.quote(string.substring(BEGINNING_OF_ELEMENT_NAME.getNbCharacter())).trim();
```

Comment:
Could you create temp variables to make this expression more readable (e.g. nodeFrom / nodeTo / tail ..)
And prefer `String.format("%s.* %s.*%s", nodeFrom, nodeTo, tail)` to string concatenation

Reply:
Done